src/definitions/subscriptionType.ts

```diff
@@ -0,0 +1,77 @@
+import { GraphQLResolveInfo } from 'graphql'
+import { ArgsValue, GetGen, MaybePromise, MaybePromiseDeep, ResultValue } from '../typegenTypeHelpers'
+import { CommonOutputFieldConfig, OutputDefinitionBlock, OutputDefinitionBuilder } from './definitionBlocks'
+import { AllNexusOutputTypeDefs } from './wrapping'
+import { NexusTypes, withNexusSymbol } from './_types'
+
+type SubscriptionTypeName = 'Subscription'
+
+export interface SubscribeFieldConfig<TypeName extends string, FieldName extends string, T = any>
+  extends CommonOutputFieldConfig<TypeName, FieldName> {
+  type: GetGen<'allOutputTypes'> | AllNexusOutputTypeDefs
+
+  subscribe(
+    root: object,
+    args: ArgsValue<TypeName, FieldName>,
+    ctx: GetGen<'context'>,
+    info: GraphQLResolveInfo
+  ): MaybePromise<AsyncIterator<T>> | MaybePromiseDeep<AsyncIterator<T>>
+
+  /**
+   * Resolve method for the field
+   */
+  resolve(
+    root: T,
+    args: ArgsValue<TypeName, FieldName>,
+    context: GetGen<'context'>,
+    info: GraphQLResolveInfo
+  ):
+    | MaybePromise<ResultValue<'Subscription', FieldName>>
+    | MaybePromiseDeep<ResultValue<'Subscription', FieldName>>
+}
```

Comment:
I know it's not really part of the pr goal, but can we maybe make `resolve` optional and provide an "identity" default value?

Reply:
Saw your comment on the PR. Ok to iterate further down the road

Reply:
Funny you say that --> https://github.com/graphql-nexus/schema/issues/78#issuecomment-654005333